unit_test/sparse/Test_Sparse_spgemm.hpp

```diff
@@ -260,22 +262,21 @@ void test_spgemm(lno_t numRows, size_type nnz, lno_t bandwidth,
   // typedef typename graph_t::entries_type::non_const_type   lno_nnz_view_t;
   // typedef typename crsMat_t::values_type::non_const_type scalar_view_t;
 
-  lno_t numCols = numRows;
   // Generate random compressed sparse row matrix. Randomly generated (non-zero)
   // values are stored in a 1-D (1 rank) array.
-  crsMat_t input_mat = KokkosKernels::Impl::kk_generate_sparse_matrix<crsMat_t>(
-      numRows, numCols, nnz, row_size_variance, bandwidth);
+  crsMat_t A = KokkosKernels::Impl::kk_generate_sparse_matrix<crsMat_t>(
+      m, k, nnz, row_size_variance, bandwidth);
+  crsMat_t B = KokkosKernels::Impl::kk_generate_sparse_matrix<crsMat_t>(
+      k, n, nnz, row_size_variance, bandwidth);
 
   crsMat_t output_mat2;
   if (oldInterface)
-    run_spgemm_old_interface<crsMat_t, device>(input_mat, input_mat,
-                                               SPGEMM_DEBUG, output_mat2);
+    run_spgemm_old_interface<crsMat_t, device>(A, B, SPGEMM_DEBUG, output_mat2);
```

Comment:
Can the old interface be removed? Has the old interface already been marked for deprecation?

Reply:
@e10harvey For now the "old" interface is not deprecated - it's just an alternative. The CrsMatrix based interface is implemented in terms of it. We test both just to be safe.

Reply:
Also there are some users that want to have close control on memory allocations that prefer the "old" interface.
That way they can create their own Kokkos Views and give them pre-allocated memory.